simavr/sim/sim_vcd_file.c

```diff
@@ -295,7 +294,7 @@ avr_vcd_init_input(
 				vcd->signal[i].size);
 		/* format is <four-character ioctl>[_<IRQ index>] */
 		if (strlen(vcd->signal[i].name) >= 4) {
-			char *dup = strdupa(vcd->signal[i].name);
+			char *dup = strdup(vcd->signal[i].name);
 			char *ioctl = strsep(&dup, "_");
```

Comment:
You're right.  What do you want the behavior of this function to be if no heap is available?  

Applying this change will make the strsep not fail, but the strlen(ioctl) that follows will fault.  I suggest "if(!dup) return -1;".   I don't expect to be out of memory in an initializer.   Another option would be to allocate a stack buffer large enough for a signal name (seems like 9 bytes =  4 (ioctl) + 1 (_) + 3 (irq index) + 1 (null) would be enough given the comment).